library/azure_rm_hdinsightcluster.py

```diff
@@ -0,0 +1,575 @@
+            - 'premium'
+    cluster_definition:
+        description:
+            - The cluster definition.
+        suboptions:
+            blueprint:
+                description:
+                    - The link to the blueprint.
+            kind:
+                description:
+                    - The type of cluster.
+            component_version:
+                description:
+                    - The versions of different services in the cluster.
+            configurations:
+                description:
+                    - The cluster configurations.
+    security_profile:
+        description:
+            - The security profile.
+        suboptions:
+            directory_type:
+                description:
+                    - The directory type.
+                choices:
+                    - 'active_directory'
+            domain:
+                description:
+                    - "The organization's active directory domain."
+            organizational_unit_dn:
+                description:
+                    - The organizational unit within the Active Directory to place the cluster and service accounts.
+            ldaps_urls:
+                description:
+                    - The LDAPS protocol URLs to communicate with the Active Directory.
+                type: list
+            domain_username:
+                description:
+                    - The I(domain) user account that will have admin privileges on the cluster.
+            domain_user_password:
+                description:
+                    - The I(domain) admin password.
+            cluster_users_group_dns:
+                description:
+                    - Optional. The Distinguished Names for cluster user groups
+                type: list
+            aadds_resource_id:
+                description:
+                    - "The resource ID of the user's Azure Active Directory I(domain) Service."
+            msi_resource_id:
+                description:
+                    - "User assigned identity that has permissions to read and create cluster-related artifacts in the user's AADDS."
+    compute_profile_roles:
+        description:
+            - The list of roles in the cluster.
+        type: list
+        suboptions:
+            name:
+                description:
+                    - The name of the role.
+            min_instance_count:
+                description:
+                    - The minimum instance count of the cluster.
+            target_instance_count:
+                description:
+                    - The instance count of the cluster.
+            vm_size:
+                description:
+                    - The size of the VM
+            linux_profile:
+                description:
+                    - The Linux OS profile.
+            virtual_network_profile:
+                description:
+                    - The virtual network profile.
+                suboptions:
+                    id:
```

Comment:
suppose this should follow common practice when referencing vnet, it could be azure resource id, it could be resource group + name, it could be name only in same resource group